File: API-JDK/module-items-bc/src/main/java/com/financialplanner/moduleitemsbc/infrastructure/jpa/Item.java
package com.financialplanner.moduleitemsbc.infrastructure.jpa;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDate;
import java.util.UUID;

@lombok.Data
@Entity
@Table(name = "Items", schema = "Item-Detail")
public class Item {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private UUID UserId;
    private String Name;
    private @NotNull Double Amount;
    private @NotNull Integer FkItemType;
    private @NotNull Integer FkPeriod;
    private @NotNull LocalDate BeginDate;
    private @NotNull LocalDate EndDate;
    private @NotNull Integer WeeklyDow;
    private @NotNull Integer EverOtherWeekDow;
    private @NotNull Integer BiMonthlyDay1;
    private @NotNull Integer BiMonthlyDay2;
    private @NotNull Integer MonthlyDom;
    private @NotNull Integer Quarterly1Month;
    private @NotNull Integer Quarterly1Day;
    private @NotNull Integer Quarterly2Month;
    private @NotNull Integer Quarterly2Day;
    private @NotNull Integer Quarterly3Month;
    private @NotNull Integer Quarterly3Day;
    private @NotNull Integer Quarterly4Month;
    private @NotNull Integer Quarterly4Day;
    private @NotNull Integer SemiAnnual1Month;
    private @NotNull Integer SemiAnnual1Day;
    private @NotNull Integer SemiAnnual2Month;
    private @NotNull Integer SemiAnnual2Day;
    private @NotNull Integer AnnualMoy;
    private @NotNull Integer AnnualDom;
    private Boolean DateRangeReq;
}

File: UI-JSX/src/features/catalog-command/transactions/types/Item.ts
// src/features/catalog-command/transactions/types/Item.ts

// C:\Users\rickd\source\repos\FPFL-React-Java\UI-JSX\src\features\catalog-command\transactions\types\Item
export interface ItemTypeDto {
  Id: number;
  Name?: string;
  [key: string]: any;
}

export interface TimePeriodDto {
  Id?: number;
  Name?: string;
  [key: string]: any;
}

/**
 * Item model used for both request and response.
 * Matches backend ApiResponse.data shape.
 */
export interface Item {
  id?: number;
  userId: string;
  name?: string;
  amount: number;
  ItemType?: ItemTypeDto | null;
  TimePeriod?: TimePeriodDto | null;
  beginDate?: string | null;
  endDate?: string | null;
  weeklyDow?: number | null;
  everyOtherWeekDow?: number | null;
  biMonthlyDay1?: number | null;
  biMonthlyDay2?: number | null;
  monthlyDom?: number | null;
  quarterly1Month?: number | null;
  quarterly1Day?: number | null;
  quarterly2Month?: number | null;
  quarterly2Day?: number | null;
  quarterly3Month?: number | null;
  quarterly3Day?: number | null;
  quarterly4Month?: number | null;
  quarterly4Day?: number | null;
  semiAnnual1Month?: number | null;
  semiAnnual1Day?: number | null;
  semiAnnual2Month?: number | null;
  semiAnnual2Day?: number | null;
  annualMoy?: number | null;
  annualDom?: number | null;
  dateRangeReq?: boolean | null;

  // POST/PUT fields expected by backend
  fkItemType?: number;
  fkPeriod?: number | null;
}
